test/Tests/src/Data/Map_Spec.enso

```diff
@@ -267,6 +266,22 @@ spec =
             m = Map.empty . insert 0 0 . insert 3 -5 . insert 1 2
             m.to_vector.should_equal [[0, 0], [3, -5], [1, 2]]
 
+        Test.specify "should allow building the map from two vectors" <|
+            expected = Map.empty . insert 0 0 . insert 3 -5 . insert 1 2
+            Map.from_keys_and_values [0, 3, 1] [0, -5, 2] . should_equal expected
+
+        Test.specify "should allow building the map from vector like things" <|
+            expected = Map.empty . insert 0 0 . insert 1 -5 . insert 2 2
+            Map.from_keys_and_values (0.up_to 3) [0, -5, 2] . should_equal expected
+
+        Test.specify "should not allow building with duplicate keys unless explicitly allowed" <|
+            expected = Map.empty . insert 0 0 . insert 3 -5 . insert 1 2
+            Map.from_keys_and_values [0, 3, 1, 0] [3, -5, 2, 0] . should_fail_with Illegal_Argument
+            Map.from_keys_and_values [0, 3, 1, 0] [3, -5, 2, 0] error_on_duplicates=False . should_equal expected
+
+        Test.specify "should not allow different length vectors when building" <|
+            Map.from_keys_and_values [0, 3, 1] [3, -5, 2, 0] . should_fail_with Illegal_Argument
```

Comment:
IMHO it feels a bit weird to use this instead of `Map.from_vector` - the keys and values are kind of 'detached' from one another.

Reply:
I liked it when combined with the Range conversions, but yes no real reason you can't use `zip` then `from_vector` just makes it a one step process.

Reply:
Hmm I'm not sure if I understand the use-case with Range conversions here. Could you please give an example?

---

Is this essentially needed because we **still** don't have the Map literal? Or is it a separate problem?

Because if it's the Map literal thing - maybe we could mark this as `UNSTABLE` and use for now but reserve the right to remove later once Map literal appears?